atlp-rwanda/cabal-phantom-backend
# #9JpkBREJ ft-register-drivers-and-operators-with proper validations

### **What does this PR do?**
requests to merge ft-register-driversd-operators-9JpkBREJ with develop
### **Description of Task to be completed?**
Created route for registering operators and drivers
Generate randomly passwords for registered users and hash them for security
Validate users
Check if a user is not registered twice
Send them email of the unhashed password
### **How should this be manually tested?**
-Visit this PR deployed app to perform below tasks https://phantom-caba-ft-registe-doghsw.herokuapp.com/api-docs/
-Clone this repo

-git checkout origin/ft-register-drivers-and-operators-9JpkBREJ

-install all packages

-make sure your db is connected with the help of .env.example file
-log in your email using help of .env.example file
**N.B:** two emails are required: the one to send password in auth function and the one to receive it
please use .env.example

-do npm run migrate

-go to your postman or browse http://localhost:5000/api-docs/
POST login using these to have admin permissions :**email:admin@phantom.com** and **password:admin**
POST at http://localhost:5000/api/v1/users/ to send an email to a new user and register them
5.Use one of the users registered password to login
use the token to write into headers  
_"Bearer **token**"_  as shown in below screenshot
### **What are the relevant pivotal tracker stories?**

[9JpkBREJ](https://trello.com/c/9JpkBREJ/10-register-drivers-operators
### **Screenshots (if appropriate)**
<img width="1440" alt="Screen Shot 2020-10-15 at 15 32 06" src="https://user-images.githubusercontent.com/68019992/96160719-a3fe9600-0f16-11eb-8656
<img width="1440" alt="Screen Shot 2020-10-20 at 21 47 45" src="https://user-images.githubusercontent.com/68019992/96636708-f9271700-131d-11eb-8493-962e89c4e44e.png">
-55f8f1deede2.png">
